sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -71,8 +71,13 @@ class FileStreamSource(
       Map()
     }}
 
+  /** exposed for testing */
+  private[sql] var clockForRetention: Clock = new SystemClock
```

Comment:
Let's remove `private[sql]` because `execution` is already a private package (SPARK-16964)